pkg/controller/certificates/approver/sarapprove.go

```diff
@@ -192,9 +194,23 @@ var kubeletServerUsages = []capi.KeyUsage{
 }
 
 func isSelfNodeServerCert(csr *capi.CertificateSigningRequest, x509cr *x509.CertificateRequest) bool {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.RotateKubeletServerCertificate) {
```

Comment:
if you want to gate it in the apiserver, gate adding isSelfNodeServerCert in `recognizers()`, and you can unit test isSelfNodeServerCert without messing with feature flag enablement in `init()`